core/src/main/java/org/apache/brooklyn/policy/InvokeEffectorOnSensorChange.java

```diff
@@ -67,7 +67,7 @@
             .build();
 
     private AttributeSensor<Object> sensor;
-
+    private Object lastValue;
```

Comment:
This won't survive rebind and I'm not sure it's thread safe (it might be - I'd need to dig into the parent's code)

Reply:
Agree, should be synchronised. I completely forgot about rebind!

Reply:
Line 82 addresses rebind actually. That is where we know the initial value to ocmpare to new from events.

Reply:
08aa7c3447658c5c6e5f518bca631b33526fa9d5

Reply:
i think this will be included in rebind anyway (even without being set in `setEntity`).  you could mark it `transient` to exclude it from rebind.